### DISCUSSION TITLE
Freezing Client State

### DISCUSSION THREAD

**Discussion by ibrizsabin** [icon-project/IBC-Integration#175] [General]
We would want to discuss the possible routes that we can take to freeze and unfreeze clients and also its consequences here. This is for the icon lightclient that will be running on archway network. The key challenge being the fact that we have to maintain consensus state at every height for btp blocks so once client is frozen we will be missing consequent consensus states. Please suggest some ideas here.

**Comment by pragyanshrestha-ibriz** [comment]
@Sid-hugobyte @MuhammedIrfan @shreyasbhat0

**Reply by Mr-Nobody21** [reply]
@pragyanshrestha-ibriz Let me check with the team as @shreyasbhat0 is "ooo" today and get back to you
